File: PE13/PE13.cpp
#include "pch.h"
#include <iostream>
#include <fstream>
#include <string>
#include "BigInteger.h"
using namespace std;

int main()
{
	ifstream sc ("PE13.txt");
	BigInteger intlist[100];
	for (int i = 0; i < 100; i++) {
		string readnum;
		getline(sc, readnum);
		BigInteger temp(readnum.substr(0, 50));
		intlist[i] = temp;
	}
	BigInteger mysum("0");
	for (BigInteger i : intlist) {
		mysum = mysum.add(i);
	}
	cout << mysum.toString().substr(0, 10) << endl;
	sc.close();
}

File: PE13/PE13.java
import java.io.*;
import java.math.*;
import java.util.*;


public class PE13 {
	public static void main(String args[]) throws IOException {
		Scanner sc = new Scanner(new File("PE13.txt"));
		ArrayList<BigInteger> intlist = new ArrayList<BigInteger>();
		for (int i = 0; i < 100; i++) {
			intlist.add(sc.nextBigInteger());
		}
		BigInteger mysum = new BigInteger("0");
		for (BigInteger i : intlist) {
			mysum = mysum.add(i);
		}
		System.out.println(mysum.toString().substring(0, 10));
		sc.close();
	}
}
